### PR TITLE
Allow to use unix socket for non-default servers

### PR DESCRIPTION
**Pull request by Totktonada**

### CODE REVIEW

File: lib/preprocessor.py [hunk 1]
```diff
@@ -211,6 +211,7 @@ def server_create(self, ctype, sname, opts):
         if 'rpl_master' in opts:
             temp.rpl_master = self.servers[opts['rpl_master']]
         temp.vardir = self.suite_ini['vardir']
+        temp.use_unix_sockets = self.suite_ini.get('use_unix_sockets', False)
```
Comment (sergw) [inline lib/preprocessor.py]: If ini contains `use_unix_sockets = False` the flag will be `True`

Comment (Totktonada) [reply lib/preprocessor.py]: Thanks, I”ll fix it.

**Comment by Totktonada** [comment]
@sergw I force-pushed the update. I touched your boolean suite.ini options parsing code, so can you look again into that, please?

File: lib/test_suite.py [hunk 2]
```diff
@@ -97,8 +110,9 @@ def __init__(self, suite_path, args):
                 dict.fromkeys(self.ini[i].split())
                 if i in self.ini else dict())
 
-        self.ini.update(
-            dict(show_reproduce_content=self.show_reproduce_content()))
+        self.parse_bool_opt('use_unix_sockets', False)
+        self.parse_bool_opt('is_parallel', False)
+        self.parse_bool_opt('show_reproduce_content', False)
```
[the remark this answers is not in the document]
Comment (Totktonada) [reply lib/test_suite.py]: Ouch, thanks! I’ll fix soon.

[the remark this answers is not in the document]
Comment (Totktonada) [reply lib/test_suite.py]: Fixed.

**Pull request merged**